lib/spack/spack/ci.py

```diff
@@ -592,6 +619,13 @@ def generate_gitlab_ci_yaml(env, print_summary, output_file,
                 variables = {}
                 if 'variables' in runner_attribs:
                     variables.update(runner_attribs['variables'])
+                    for name, value in variables.items():
+                        m = runner_var_regex.search(value)
```

Comment:
As an alternative to my above comment about the regex:

Consider using `runner_var_regex.match(value)` instead of `search()`.  This change should also have the effect of requiring that the match occur at the very beginning of `value`.